fed/_private/global_context.py

```diff
@@ -28,27 +35,40 @@ def next_seq_id(self) -> int:
     def get_cleanup_manager(self) -> CleanupManager:
         return self._cleanup_manager
 
-    def job_name(self) -> str:
+    def get_job_name(self) -> str:
         return self._job_name
 
+    def get_failure_handler(self) -> Callable[[], None]:
+        return self._failure_handler
+
+    def acquire_shutdown_flag(self) -> bool:
```

Comment:
It's too weird to pass the method to cleanup manager. A better practice I believe is implementing it in cleanup manager and then use it in anywhere.

Reply:
I understand. The reason to do so is that: 
The lock has to be maintained in `GlobalContext` , because it's indeed a part of the job context;
, but directly accessing it in `CleanupManager`  needs to import "global_context" which causes a cross-import.